src/muqsit/chestshop/EventHandler.php

```diff
@@ -49,9 +49,10 @@ public function handleTransaction(Player $player, Item $itemClicked, Item $itemC
 
 		if($nbt->hasTag("Button")){
 			$currentpage = $this->currentpage[$playerId = $player->getId()] ?? 1;
+			$inventoryAction->getInventory()->clearAll();//A little hack to prevent strange behaviours in mcbe 1.4+(if the second page wasn't full, items from previous page were displayed with items in the new page. Reference: https://imgur.com/a/KwOrOGi
```

Comment:
It's more than a "little" hack, this will cause even more issues.
The correct way to fix this issue is fixing it where you fixed the buttons.
After pushing the buttons into the `$contents` array, use `Inventory::setContents($contents);`.